README.md

```diff
@@ -1 +1,174 @@
+
+```js
+var banana = require('path')
+console.log(banana.parse('/home/user/azat/program.js'))
+```
+
+Let's cover the most important core modules.
+
+Note: the code is a simplified version from the Node.js GitHub repository where core modules are located. The file is `path.js` in the `lib` folder, in case you want to see the real implementation.
+
+## Main Core Modules
+
+There are a lot of core modules but not as many compare to other platforms/environments like Python. The philosophy of Node is to keep the core lean and thin. But because Node was build for networking (web app, clients and servers) with mind, the creators of Node bundled it with some great module. Really you can build your entire HTTP or IRC server with just core modules!
+
+Here is the list of the core modules for v5.1:
+
+* [`assert`](https://nodejs.org/api/assert.html): A simple set of assertion test
+* [`cluster`](https://nodejs.org/api/cluster.html): Interface to launch a cluster (multiple) of Node processes to increase performance 
+* [`crypto`](https://nodejs.org/api/crypto.html): Cryptographic functionality for OpenSSL, HMAC, etc.
+* [`events`](https://nodejs.org/api/events.html): Interface for asynchronous event-driven architecture (used by many other core modules)
+* [`fs`](https://nodejs.org/api/fs.html): File system I/O methods
+* [`http`](https://nodejs.org/api/http.html): Interface for HTTP servers and clients
+* [`net`](https://nodejs.org/api/net.html): Interface for networking which can be used to implement any protocol, not just HTTP (`http` uses `net`)
+* [`os`](https://nodejs.org/api/os.html): Basic operating-system utilities
+* [`path`](https://nodejs.org/api/path.html): Utilities for handling and transmitting file paths on various platforms.
+* [`querystring`](https://nodejs.org/api/querystring.html): Utilities for working with query strings (e.g., `?key1=value1&key2=value2` in a URL)
+* [`stream`](https://nodejs.org/api/stream.html): Abstract interface implemented by various objects.
+* [`url`](https://nodejs.org/api/url.html): Utilities for URL resolution and parsing.
+* [`util`](https://nodejs.org/api/util.html): Utilities which are mostly used by other core Node modules
+
+Are your head is about to explode from this list of core modules? Stop the panic. You don't need to remember all of them. Just know that they are out there when you have a task at hand which requires you to require that module (pun intended), e.g., `crypto` for hashing of the passwords.
+
+In the next section, we'll show you some examples of `fs`, `path`, and `url`.
+
+## fs
+
+You can read a file asynchronously with `fs.readFile` and write to a file with `fs.writeFile`, e.g.,
+
+```js
+var callback = function(error, data) {
+  console.log(data, error)
+}
+fs.readFile('data.json', 'utf8', callback)
+
```

Comment:
add a note about what the parameters are here - file name, character encoding, and callback function for what to do if it can't read the file